internal/testrunner/runners/system/runner.go

```diff
@@ -1333,6 +1376,18 @@ func (r *runner) validateTestScenario(ctx context.Context, result *testrunner.Re
 		return result.WithError(err)
 	}
 
+	// TODO: remove this once Elasticsearch will map event.original correctly (https://github.com/elastic/elasticsearch/pull/106714)
+	ignoredFields := make([]string, 0, len(scenario.ignoredFields))
+	for _, field := range scenario.ignoredFields {
+		if field != "event.original" {
+			ignoredFields = append(ignoredFields, field)
+		}
+	}
+
+	if len(ignoredFields) > 0 {
+		return result.WithError(fmt.Errorf("found ignored fields in data stream %s: %v", scenario.dataStream, ignoredFields))
```

Comment:
As I was thinking about getting someone to investigate the errors that were surfaced by this enhancement, I am realizing that this error needs more information in it. As it stands now, there isn't enough information in the error to attempt solving it.

The first thing a package developer needs to do is replicate the test setup and then run the same query to obtain information that elastic-package already has in order to see raw values from the `_source` or `ignored_field_values`. Can we create a structured error that includes this information or log the raw data in some way to improve DX?

Reply:
That's a fair point @andrewkroh - Getting the top documents with `_ignored` and making them part of the error message seems like the best way to help the user fix the problem. I will also add a note to `docs/howto/system_testing.md` to explain what's this error is about and how to debug it.

Reply:
It would be nice if this were significantly less verbose; the vast majority of the output from this is unhelpful in debugging the issue and we already have a very high noise to signal ratio here.

If only the relevant fields are emitted this becomes much more manageable. For example with this change, running system tests on m365_defender
```diff
diff --git a/internal/testrunner/runners/system/runner.go b/internal/testrunner/runners/system/runner.go
index 332925a..838da5b 100644
--- a/internal/testrunner/runners/system/runner.go
+++ b/internal/testrunner/runners/system/runner.go
@@ -21,6 +21,7 @@ import (
        "github.com/Masterminds/semver/v3"
        "gopkg.in/yaml.v3"
 
+       "github.com/elastic/elastic-agent-libs/mapstr"
        "github.com/elastic/elastic-package/internal/agentdeployer"
        "github.com/elastic/elastic-package/internal/common"
        "github.com/elastic/elastic-package/internal/configuration/locations"
@@ -1403,7 +1404,22 @@ func (r *runner) validateTestScenario(ctx context.Context, result *testrunner.Re
        }
 
        if len(ignoredFields) > 0 {
-               degradedDocsJSON, err := json.MarshalIndent(scenario.degradedDocs, "", "  ")
+               issues := make([]struct {
+                       ID            any `json:"_id"`
+                       Timestamp     any `json:"@timestamp,omitempty"`
+                       IgnoredFields any `json:"ignored_field_values"`
+               }, len(scenario.degradedDocs))
+               for i, d := range scenario.degradedDocs {
+                       issues[i].ID = d["_id"]
+                       switch s := d["_source"].(type) {
+                       case map[string]any:
+                               issues[i].Timestamp = s["@timestamp"]
+                       case mapstr.M:
+                               issues[i].Timestamp = s["@timestamp"]
+                       }
+                       issues[i].IgnoredFields = d["ignored_field_values"]
+               }
+               degradedDocsJSON, err := json.MarshalIndent(issues, "", "  ")
                if err != nil {
                        return nil, fmt.Errorf("failed to marshal degraded docs to JSON: %w", err)
                }
```
we get something like this
```
Error: error running package system tests: could not complete test run: found ignored fields in data stream logs-m365_defender.incident-ep: [m365_defender.incident.alert.recommended_actions]. Affected documents: [
  {
    "_id": "RJDPoPySgbgCrqqObuC4ic2x9QE=",
    "@timestamp": "2021-09-30T09:35:45.113Z",
    "ignored_field_values": {
      "m365_defender.incident.alert.recommended_actions": [
        "Collect artifacts and determine scope\n�\tReview the machine timeline for suspicious activities that may have occurred before and after the time of the alert, and record additional related artifacts (files, IPs/URLs) \n�\tLook for the presence of relevant artifacts on other systems. Identify commonalities and differences between potentially compromised systems.\n�\tSubmit relevant files for deep analysis and review resulting detailed behavioral information.\n�\tSubmit undetected files to the MMPC malware portal\n\nInitiate containment \u0026 mitigation \n�\tContact the user to verify intent and initiate local remediation actions as needed.\n�\tUpdate AV signatures and run a full scan. The scan might reveal and remove previously-undetected malware components.\n�\tEnsure that the machine has the latest security updates. In particular, ensure that you have installed the latest software, web browser, and Operating System versions.\n�\tIf credential theft is suspected, reset all relevant users passwords.\n�\tBlock communication with relevant URLs or IPs at the organization�s perimeter."
      ]
    }
  }
]
```